CEXT-2856: Add context to event provider in onboarding docs (#218)
AdobeDocs/commerce-extensibility: src/pages/events/configure-commerce.md

Reviewer: @jhadobe we've gotten feedback from multiple customers that they don't understand how to confirm that they completed each step correctly, especially with the Commerce configuration step. Often times they expect to be able to see the event provider in the Developer Console project. They get confused because they are not sure if they plugged in all the values correctly. Can we find a way to tell a user what to expect when they finish the Complete the Commerce configuration step? Maybe something like "you will not be able to see the event provider in the Developer Console UI until you subscribe to a specific event"

Reviewer: partner issue "Nothing shows up until you add an event registration to the developer console - need to update configure-commerce docs to help developer understand what the create-event-provider command does"

Reviewer 2: good feedback. i've added a suggestion below @naomirobichaud

> 1. Copy the contents of the `<workspace-name>.json` file into the **Adobe I/O Workspace Configuration** field.

Reviewer: @jhadobe can we change this to something like "entire contents". Let's make it easier to understand that this means the entire file, as it can be a large file with several fields. This step is a bit confusing bc all the other copy/paste config steps focus on 1 particular field/value.

Reviewer 2: @naomirobichaud in my opinion "entire contents" would be redundant, but i do not think it harms anything to modify the sentence.

Reviewer: Sounds good - IMO it's worth specifying

> 1. Copy the contents of the `<workspace-name>.json` file into the **Adobe I/O Workspace Configuration** field.

Reviewer 2: ```suggestion
1. Copy the entire contents of the `<workspace-name>.json` file into the **Adobe I/O Workspace Configuration** field.
```

> 1. Enter a unique identifier in the **Adobe Commerce Instance ID** field. This value identifies your project and routes events from Adobe Commerce to the correct `Event Provider` in Adobe I/O. The value of this field is visible in App Builder during the [creation of Event Registration](#subscribe-and-register-events).

Reviewer: @oshmyheliuk this value is just an identifier - so I think it may be helpful to change the language in the second sentence to be more clear. @jhadobe what do you think about the use of "routes" and "project" here. IMO project is a bit confusing here bc we're already using "project" as an App Builder concept in these docs. "Routes" makes it sound like a key or auth mechanism rather than just a descriptive name.

Maybe we instead use language like "This value identifies this particular Commerce Instance as a publisher of events. It is important to use a descriptive name in case you have multiple Commerce Instances sending events to App Builder. You will see this value in later configuration steps and use this name to select the correct event provider. The value of this field is visible in App Builder during the..." 

I actually like this screenshot bc it helps visualize where you'll use this name. What do you think about using something like this @jhadobe 
https://files.slack.com/files-pri/T23RE8G4F-F075D31RQ21/screenshot_2024-05-28_at_12.41.51___pm.png

Reviewer: I say all this bc we've gotten a lot of feedback on these configuration fields - people are often very confused about their purpose. 

Andy confused about what an event provider is - Needs to understand what the audience is / who the labels are for. See Rightpoint feedback - https://wiki.corp.adobe.com/display/~nrobichaud/App+Builder+Partner+Feedback+Parking+Lot

Reviewer 2: it is possible that the using the term "project" could make a dev think that they need the PROJECTID from App Builder. what if we reword to:

 ```suggestion
1. Enter a unique identifier in the **Adobe Commerce Instance ID** field. This unique value identifies your Commerce instance, which allows Commerce events to connect to the correct `Event Provider` in Adobe I/O. This ID corresponds to the **Provider** displayed when [subscribing to events](#subscribe-and-register-events).
```

>    **NOTE**: The **Adobe Commerce Instance ID** field only support alphanumeric characters, hyphens and underscores.

Reviewer: Needs to be "supports"

Author: Updated to `supports`

> 1. Click **Save Config**, but do not leave the page. The next section creates an event provider, which is necessary to complete the configuration.
> 

Reviewer 2: ```suggestion
1. Click **Save Config**, but do not leave the page. The next section creates an event provider, which is necessary to complete the configuration.

 The event provider will not appear in the Developer Console until after you subscribe to a specific event.

```

Author: Does a "specific event" mean "any event" in this context? As it requires subscribing to any at least one event to make the Event Provider shown in UI

Reviewer 2: i think so, @naomirobichaud is it any event?

Reviewer: At least 1 event - we should specify that subscribe in this case means configuring an event to be emitted on the Commerce side so via io_events.xml, config.php, CLI

## Create an Event Provider

Create an `Event Provider` in Adobe I/O Events to associate the Commerce Events subscriptions with the provider. The event subscriptions in Adobe Commerce are created as `Event Metadata` in Adobe I/O Events infrastructure.

Each `Event Provider` can link to multiple event subscriptions (`Event Metadata`). The event's subscriptions will be automatically linked to your `Event Provider` whenever you subscribe with the `events:subscribe` CLI command. You can also manually synchronize all subscriptions with the `events:metadata:populate` command.

> 

Reviewer 3: Is it worth mentioning that the synchronization is also done when `setup:upgrade` is run for cloud customers defining event subscriptions in `io_events.xml`?

Author: I added the next sentence:
```
The events subscriptions synchronization is also called each time during the execution of `setup:upgrade` command.
```
